tests/integration/functional/zen_stores/test_zen_store.py

```diff
@@ -377,6 +381,161 @@ def test_deleting_default_workspace_fails():
 # '-------'
 
 
+class TestAdminUser:
+    default_pwd = "".join(random.choices(ascii_lowercase, k=10))
+    test_user = "test_user"
+
+    @contextmanager
+    def _login(
+        self,
+        zen_store: RestZenStore,
+        user_name: str,
+        password: str,
+    ):
+        zen_store.clear_session()
+        orig_user_name = zen_store.config.username
+        zen_store.config.username = user_name
+        orig_password = zen_store.config.password
+        zen_store.config.password = password
+        zen_store.session
+
+        try:
+            yield
+        finally:
+            zen_store.clear_session()
+            zen_store.config.username = orig_user_name
+            zen_store.config.password = orig_password
+            zen_store.session
+
+    def test_creation_as_admin_and_non_admin(self):
+        """Tests creating a user as an admin and as a non-admin."""
+        if Client().zen_store.type == StoreType.SQL:
+            pytest.skip("SQL ZenStore does not support admin users.")
+
+        with UserContext(password=self.default_pwd) as test_user:
+            zen_store: RestZenStore = Client().zen_store
+            # this is not allowed for non-admin users
+            with self._login(
+                zen_store, user_name=test_user.name, password=self.default_pwd
+            ):
+                with pytest.raises(IllegalOperationError):
+                    zen_store.create_user(
+                        UserRequest(
+                            name=self.test_user + "2",
+                            password=self.default_pwd,
+                        )
+                    )
```

Comment:
```suggestion
        with UserContext(login=True):
            zen_store: RestZenStore = Client().zen_store
            # this is not allowed for non-admin users
            with pytest.raises(IllegalOperationError):
                zen_store.create_user(
                    UserRequest(
                        name=sample_name(self.test_user),
                        password=self.default_pwd,
                    )
                )
```

That's what I meant when I said `UserContext(login=True)` can completely replace the `_login` context manager you created here.